flight/Libraries/math/lqg.c

```diff
@@ -160,10 +179,17 @@ struct rtkf_state {
  void rtkf_stabilize_covariance(rtkf_t rtkf, int iterations)
  {
  	PIOS_Assert(rtkf);
- 	for (int i = 0; i < iterations; i++) {
- 		rtkf_calculate_covariance_3x3(rtkf->A, rtkf->K, rtkf->P, rtkf->Q, rtkf->R);
+
+	if (rtkf->solver_iterations < 0)
+		return;
+
+	for (int i = 0; i < iterations; i++) {
+		rtkf->solver_iterations++;
+		if (rtkf->solver_iterations > SOLVER_MAX || (rtkf_calculate_covariance_3x3(rtkf->A, rtkf->K, rtkf->P, rtkf->Q, rtkf->R) && rtkf->solver_iterations > SOLVER_MIN)) {
+			rtkf->solver_iterations = -1;
```

Comment:
It sure would be nice if "reached max" was not considered "solved" here.

Reply:
How to handle this? Warning but still let them arm?

Reply:
That would be one way-- or refuse to allow arm if they have any LQG flight modes and things are unsettled (sanity check), or whatever.  Just about anything is better than silently using known-bad coeffs.